components/modelmeshserving/modelmeshserving.go

```diff
@@ -33,31 +33,34 @@ type ModelMeshServing struct {
 
 func (m *ModelMeshServing) OverrideManifests(_ string) error {
 	// Go through each manifests and set the overlays if defined
-	for _, subcomponent := range m.DevFlags.Manifests {
-		if strings.Contains(subcomponent.URI, DependentComponentName) {
-			// Download subcomponent
-			if err := deploy.DownloadManifests(DependentComponentName, subcomponent); err != nil {
-				return err
-			}
-			// If overlay is defined, update paths
-			defaultKustomizePath := "base"
-			if subcomponent.SourcePath != "" {
-				defaultKustomizePath = subcomponent.SourcePath
+	if m.DevFlags != nil && len(m.DevFlags.Manifests) != 0 {
```

Comment:
len() check is redundant here (the same in many other places, but it was there before the patch). I personally would make  if m.DevFlags == nil { return nil} to avoid this extra indentation change and make intentions clear, but looks like it does not fit the style.

Reply:
Previously it was just `if len(m.DevFlags.Manifests) != 0 `
but with the addition of the `*DevFlags`, while the unmarshalling happened, it was a `nil` field and trying to access a `nil` field like `m.DevFlags.Manifests` creates a error(here DevFlags being `nil`). So I added the check.

Reply:
removed the redundant len checks now and returned nil if DevFlags is nil early in the code :)